### PR TITLE
Fix broken links to glossary items in user guide

### PR DESCRIPTION
**Pull request by edgarrmondragon**
This PR fixes some links to concepts/glossary in the user guide

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
The failing test seems to be a flaky one: https://github.com/agronholm/apscheduler/actions/runs/8063354336/job/22024980072?

File: docs/userguide.rst [hunk 1]
```diff
@@ -103,24 +103,24 @@ even some external service.
...
 them, and then sending back the results to the `data store`_.
 
 Running the scheduler
 =====================
 
 The scheduler_ comes in two flavors: synchronous and asynchronous. The synchronous
 scheduler actually runs an asynchronous scheduler behind the scenes in a dedicated
-thread, so if your app runs on :mod:`asyncio` or Trio_, you should prefer the
-asynchronous scheduler.
+thread, so if your app runs on :mod:`asyncio` or `Trio <https://trio.readthedocs.io/en/stable/>`_,
```
Comment (agronholm) [inline docs/userguide.rst:122]: Just add a counterpart to this later (`.. _Trio: https://trio.readthedocs.io/en/stable/`). I dislike inline links.

**Review by agronholm — CHANGES_REQUESTED** [review]
Looks mostly good, just one change I'd like.

File: docs/userguide.rst (hunk 1 above)
Comment (edgarrmondragon) [reply docs/userguide.rst]: fa0dba554264007896abd64fc42429e0fe658a47

**Review by agronholm — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks!

**Pull request merged**